### PR TITLE
[List] Update List readme to include information on accessibility and color/…

### LINKED ISSUE #5402: [List] Update List Readme to include information on accessibility and color/typography subsystems

### PR DESCRIPTION
**Pull request by andrewoverton**
Closes #5402.

### CODE REVIEW

File: components/List/docs/README.md [hunk 1]
```diff
@@ -0,0 +1,244 @@
+# List
+
+Material Design Lists are continuous groups of text and/or images. The [Material guidelines](https://material.io/go/design-lists) for Lists are extensive, and there is no class at this time for implementing any one of them, let alone all of them. However, we are starting to add classes that represent individual List Items. We currently offer two List Item Cells:
+
```
Comment (jverkoey) [inline components/List/docs/README.md:6]: Consider adding the following doc tags:

```
<!-- badges -->

<!-- design-and-api -->

<!-- toc -->
```

Comment (jverkoey) [reply components/List/docs/README.md:6]: Example: https://raw.githubusercontent.com/material-components/material-components-ios/develop/components/AppBar/docs/README.md

File: components/List/docs/accessibility.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
+## Accessibility
+
+To help ensure your Lists are accessible to as many users as possible, please be sure to reivew the following
```
Comment (jverkoey) [inline components/List/docs/accessibility.md]: Tip: s/reivew/review/

File: components/List/docs/README.md [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+- [Typical use](typical-use.md)
+
+## Extensions
+
+- [Color Theming](color-theming.md)
+- [Typography Theming](typography-theming.md)
+
+- [Accessibility](accessibility.md)
+
+## How to implement your own List Cell
```
Comment (jverkoey) [inline components/List/docs/README.md:43]: Consider extracting this to a separate document and linking to it in the list of articles above.

File: components/List/docs/color-theming.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+```objc
+// Step 1: Import the ColorThemer extension
+#import "MaterialList+ColorThemer.h"
+
+// Step 2: Create or get a color scheme
+id<MDCColorScheming> colorScheme = [[MDCSematnicColorScheme alloc] init];
+
+// Step 3: Apply the color scheme to your component from within `-collectionView:cellForItemAtIndexPath:`
+[MDCListColorThemer applySemanticColorScheme:colorScheme
+                                 toBaseCell:cell];
```
Comment (jverkoey) [inline components/List/docs/color-theming.md]: Tip: alignment of arguments.

**Review by jverkoey — APPROVED** [review]
Lots going on in this PR - possibly could have been broken up into smaller pieces but given that this is documentation there's not a high risk of needing to roll back any part of this CL, so let's leave it as-is.

File: components/List/docs/README.md (hunk 1 above)
Comment (andrewoverton) [reply components/List/docs/README.md:6]: Done.

File: components/List/docs/README.md (hunk 3 above)
Comment (andrewoverton) [reply components/List/docs/README.md:43]: Done.

File: components/List/docs/accessibility.md (hunk 2 above)
Comment (andrewoverton) [reply components/List/docs/accessibility.md]: Done

File: components/List/docs/color-theming.md (hunk 4 above)
Comment (andrewoverton) [reply components/List/docs/color-theming.md]: Done

**Review by jverkoey — APPROVED** [review]

**Comment by jverkoey** [comment]
```
Broken link schemes/Typography in list/README.md
```

File: components/List/docs/create-your-own.md [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+
+Note: It is better to set the size smaller rather than larger or constraints might break in runtime.
+
+```objc
+_flowLayout.estimatedItemSize = CGSizeMake(kSmallArbitraryCellWidth, kSmallestCellHeight);
+```
+
+### Typography
+
+For our example we use a typography scheme to apply the fonts to our cell's `UILabel`'s. Please see <a href="../schemes/Typography">Typography Scheme</a> for more info.
```
Comment (jverkoey) [inline components/List/docs/create-your-own.md]: This likely needs to be `../../schemes/Typography` now that we're in the docs folder.

Comment (andrewoverton) [reply components/List/docs/create-your-own.md]: Thanks!

**Pull request merged**